btree/btree.c

```diff
@@ -3740,6 +3761,61 @@ int64_t btree_open_tick(struct m0_sm_op *smop)
 	}
 }
 
+/**
+ * btree_close_tree_tick function is used to traverse through different states
+ * to facilitate the working of m0_btree_close().
+ *
+ * @param smop     represents the state machine operation
+ * @return int64_t returns the next state to be executed.
+ */
+int64_t btree_close_tree_tick(struct m0_sm_op *smop)
+{
+	struct m0_btree_op    *bop     = M0_AMB(bop, smop, bo_op);
+	struct td             *td_curr = bop->bo_arbor->t_desc;
+	struct nd             *nd_head = ndlist_tlist_head(&td_curr->
+							   t_active_nds);
+
+	switch (bop->bo_op.o_sm.sm_state) {
+	case P_INIT:
+		M0_ASSERT(td_curr->t_ref != 0);
+		if (td_curr->t_ref > 1)
+			tree_put(td_curr);
+		else if (td_curr->t_ref == 1) {
+			if (td_curr->t_starttime == 0)
+				td_curr->t_starttime = m0_time_now();
+
+			if (ndlist_tlist_length(&td_curr->t_active_nds) > 1)
+				return P_TIMECHECK;
+			else
+				return P_ACT;
+		}
+
+	case P_TIMECHECK:
+		/**
+		 * This code is meant for debugging. In future, this case needs
+		 * to be handled in a better way.
+		 */
+		if (m0_time_seconds(m0_time_now() - td_curr->t_starttime) > 5) {
+			td_curr->t_starttime = 0;
+			return M0_ERR(-ETIMEDOUT);
+		}
+		if (ndlist_tlist_length(&td_curr->t_active_nds) > 1)
+			return P_TIMECHECK;
+		return P_ACT;
```

Comment:
You can remove `return P_ACT` so that  it will falldown.

Reply:
P_DONE